### ISSUE TITLE
Show error when deleting inbound line that is already used

### ISSUE THREAD

**Issue by richardmoizeau** [openmsupply/open-msupply#1807]
See last comment

## What went wrong? 😲
 

1. I created a new inbound shipment for one item: 
<img width="1072" alt="Screenshot 2023-05-20 at 2 11 26 PM" src="https://github.com/openmsupply/open-msupply/assets/74992958/bf2c6ebe-6f01-4c16-a878-f3abc57c68fc">

2. I confirmed the inbound shipment as "delivered"

3. When looking for the stock line in "View Stock", I can already see it: 
<img width="1245" alt="Screenshot 2023-05-20 at 2 11 43 PM" src="https://github.com/openmsupply/open-msupply/assets/74992958/7d733748-8108-4915-a21a-6e75861e4d1a">

## Expected behaviour 🤔

I think we should confirm the stock movement (goods in) when shipment has been verified ?

## How to Reproduce 🔨

see What went wrong? 😲

## Your environment 🌱

 - Version: 1.1.13
 - Platform:
   - [ ] desktop (windows)
   - [ ] desktop (macOS)

**Comment by richardmoizeau** [comment]
@DhanyaHerath @andreievg, maybe this is not a bug ?

**Comment by andreievg** [comment]
This is the same functionality as mSupply Supplier Invoices (confirmed = stock is in, finalised = nothing really changes), there would be an error if you try to update stock in supplier invoice if it has been confirmed and was already used in invoice

Btw I also thought it was confusing but was asked to follow how mSupply does it.

I just realised there would be another error, if this stock line was used somewhere else ?

Create inbound shipment with stock and confirm delivery.

Add stocktake with that stock line and adjust stock.

Try to delete this line in inbound shipment, there would be an error, but you can't see it and that line remains. This is the error:

```

 
"DatabaseError(\n    ForeignKeyViolation(\n        \"\\\"update or delete on table \\\\\\\"stock_line\\\\\\\" violates foreign key constraint \\\\\\\"stocktake_line_stock_line_id_fkey\\\\\\\" on table \\\\\\\"stocktake_line\\\\\\\"\\\"\",\n    ),\n)"
input
: 
"DeleteInboundShipmentLine {\n    id: \"7132d148-3325-471b-8dd5-c3e0fcd38a38\",\n}"
locations
: 
[{line: 2, column: 3}]
message
: 
"Internal error"
path
: 
["batchInboundShipment"]
```

Also need to show that there is an error when doing batch, this should be done in the new issue (issue created before this issue is considered fixed/merged)

**Comment by andreievg** [comment]
I think minimum change, when deleting this stocktake line, show error message

**Comment by nisha-dangol** [comment]
Tested in v1.4.00-test-2023-10-02-build exe

## Tests
- [x] Create a manual inbound with any item
- [x] Confirm delivered (this will create a stock line for the item)
- [x] Create a stocktake with said item and change stock (change counted packs)
- [x] Go back to the manual inbound invoice that was created and try to delete line
- [ ] See snack **- FAILED- I don't see any snack nor toast alerts for that.** 
@roxy-dao See [this recording](https://recordit.co/zDgPlLyaq5) reference.

**Comment by nisha-dangol** [comment]
Rechecked in v1.5.4. I still see the error of no correct snack, or should I say, in v1.5.4, I'm seeing `Line deleted` snack/toast but the line's not deleted. 
Can be discussed/addressed in new issue now I guess: #2669
